wsmaster/che-core-api-devfile/src/test/java/org/eclipse/che/api/devfile/server/convert/ProjectConverterTest.java

```diff
@@ -131,9 +139,59 @@ public void testCloningIntoProjectsRootFails() throws Exception {
     ProjectImpl devfileProject =
         new ProjectImpl(
             "myProject",
-            new SourceImpl("git", "https://github.com/eclipse/che.git", "master"),
+            new SourceImpl("git", "https://github.com/eclipse/che.git", "master", null, null, null),
             "not/../in/root/../..");
 
     projectConverter.toWorkspaceProject(devfileProject);
   }
+
+  @Test(expectedExceptions = DevfileException.class)
+  public void testOnlyOneOfStartPointTagAllowed() throws Exception {
+    ProjectImpl devfileProject =
+        new ProjectImpl(
+            "myProject",
+            new SourceImpl("git", "https://github.com/eclipse/che.git", null, "a", "b", null),
+            null);
+
+    projectConverter.toWorkspaceProject(devfileProject);
+  }
+
+  @Test(expectedExceptions = DevfileException.class)
+  public void testOnlyOneOfStartPointCommitIdAllowed() throws Exception {
+    ProjectImpl devfileProject =
+        new ProjectImpl(
+            "myProject",
+            new SourceImpl("git", "https://github.com/eclipse/che.git", null, "a", null, "b"),
```

Comment:
Maybe it would be better to cover all combination in a single test with dataProvider. WDYT?